### ISSUE TITLE
If consent is not granted, I can't send any traces, even those without any user information.

### ISSUE THREAD

**Issue by MaximBazarov** [DataDog/dd-sdk-ios#1716]
### Question

Hi, I'm truing to send our internal traces to Datadog, and I can see if user is not giving a consent I can't even enable tracing. That essentially makes it unusable for iOS app instrumentation. I think this should not be the case, app developers should decide what traces to send based on the legal requirement and I don't think it's a Datadog responsibility to prevent developers from that.

As a workaround I tried to add `span_id` and `trace_id` to the logs, but Datadog doesn't seem to include this logs in the traces view.

Is there a way to enable tracings for manually sent events when consent is not granted?

**Comment by MaximBazarov** [comment]
Another issue is Logger also sends no events if consent is not granted, which makes absolutely no sense, we should be able to send anonymous logs.

**Comment by MaximBazarov** [comment]
Hey @maxep thank you for the response! 

Yes that's a good workaround, it's fun how I wrote almost identical code :D, kudos on having this core as a source of truth for instruments.

I just think this something that could be an improvement to the SDK, to strip all the automatic collection/transmission of the PII context if consent to that is not granted, but still allow sending logs that initiated by the system and leave developers to make sure what they trace is not a subject.

**Comment by maxep** [comment]
Hi @MaximBazarov 

That's a fair call, I will bring this topic to the team 👍 

Apple is also adding an extra layer for privacy control and are now requiring [Privacy Manifests](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files) declaration. Performance data are subject of it, so even with looser rules on our side, reporting could be blocked by the system.
